torch/distributed/_composable/contract.py

```diff
@@ -57,96 +54,93 @@ def contract(func):
-        )
-        orig_named_modules = OrderedDict(
-            module.named_modules(remove_duplicate=False)
-        )
-
-        updated = func(module, *args, **kwargs)
-
-        if updated is None:
-            updated = module
-
-        new_named_params = OrderedDict(updated.named_parameters())
-        new_named_buffers = OrderedDict(
-            updated.named_buffers(remove_duplicate=False)
-        )
-        new_named_modules = OrderedDict(
-            updated.named_modules(remove_duplicate=False)
-        )
-
-        assert isinstance(updated, nn.Module), (
-            "Output of composable distributed APIs must be either None or "
-            f"nn.Module, but got {type(updated)}"
-        )
-
-        def check_fqn(orig_fqns: List[str], new_fqns: List[str]):
-            if orig_fqns == new_fqns:
-                return
-
-            orig_fqn_set, new_fqn_set = set(orig_fqns), set(new_fqns)
-            orig_only = orig_fqn_set - new_fqn_set
-            new_only = new_fqn_set - orig_fqn_set
-            if len(orig_only) or len(new_only):
-                raise RuntimeError(
-                    "Composable distributed API implementations cannot modify "
-                    "FQNs.\n"
-                    f"Only in original FQNs: {orig_only},\n"
-                    f"Only in new FQNs: {new_only}"
-                )
-            else:
-                raise RuntimeError(
-                    "Composable distributed API implementations cannot modify "
-                    "the order of FQNs.\n"
-                    f"Original FQNs: {orig_only}\n"
-                    f"New FQNs: {new_only}"
-                )
-
-        check_fqn(
-            list(orig_named_params.keys()), list(new_named_params.keys())
-        )
-        check_fqn(
-            list(orig_named_buffers.keys()), list(new_named_buffers.keys())
-        )
-        check_fqn(
-            list(orig_named_modules.keys()), list(new_named_modules.keys())
-        )
-
-        # TODO: a stricter verification should also reject changing module
-        # types and monkey-patching forward() method implementations.
-
-        # TODO: verify that installed distributed paradigms are compatible with
-        # each other.
-
-        return updated
-
-    def get_state(module: nn.Module) -> Optional[_State]:
-        return module.__dict__.setdefault(  # type: ignore[call-overload]
-            STATE_KEY,
-            {},  # TODO(@yhcharles): this is a temporary fix, need a better way
-        ).get(
-            func
-        )  # type: ignore[call-overload]
-
-    wrapper.state = get_state  # type: ignore[attr-defined]
-
-    return wrapper
+    def inner(func):
+        def wrapper(module: nn.Module, *args, **kwargs) -> Optional[nn.Module]:
+            # install states specific to the wrapped ``func``
+            default_all_state: Dict[Callable, _State] = {}
+            all_state: Dict[Callable, _State] = module.__dict__.setdefault(  # type: ignore[call-overload]
```

Comment:
Should we update this mapping here to use the one in `_composable_state.py`? So that we are consistently using `module -> State` mapping.